### ISSUE TITLE
Cannot Access a closed stream

### ISSUE THREAD

**Issue by hashimvellachal** [mattosaurus/PgpCore#139]
Hi ,
 i am not able decrypt file using below command , getting error Cannot Access a closed stream.

Code : 
pgp.DecryptFile(plainFilePath, plainFilePath = Path.GetDirectoryName(filePath) + @"\" + fileName + ".txt",  Server.MapPath(@"keys\PVT_KEY.gpg"), "Password");

Note: same file i am able to decrypt using kleoptra tool using same password and private key

**Comment by mattosaurus** [comment]
This should hopefully be fixed in v5.4.0

**Comment by hashimvellachal** [comment]
5.4.0 having same issue .

**Comment by robert-shattock** [comment]
Hey @mattosaurus, like @hashimvellachal I'm still getting the "Cannot access a closed stream" error before and after upgrading to 5.4.0.  Is there a workaround or ETA on a fix?

**Comment by hashimvellachal** [comment]
thank you . its working

**Comment by hashimvellachal** [comment]
> 
> 
> Hey @mattosaurus, like @hashimvellachal I'm also getting the "Cannot access a closed stream" error - I had assumed it was because I upgraded PGPCore to 5.4.0 but after some trial and error it seems the issue is Portable.BouncyCastle 1.9.0... when I downgrade Portable.BouncyCastle to 1.8.10 my decrypt code works correctly with PGPCore 5.4.0.
> 
> So the workaround is to use PGPCore 5.4.0 with Portable.BouncyCastle 1.8.10.
> 
> But obviously there is still a bug with the combination of PGPCore 5.4.0 and Portable.BounceCastle 1.9.0 (since PGPCore 5.4.0 is meant to work with any version >= 1.8.9).
> 
> (p.s. Thanks for your effort in building a great library!)

**Comment by virouzrx** [comment]
@robert-shattock I can't get it to work. I installed the 5.4.0 version of PgpCore and installed Portable.BouncyCastle in 1.8.10 version and the same problem still occurs. Can you provide a little bit more info of how did you manage to get it to work?

**Comment by virouzrx** [comment]
Thanks for the quick reply. I checked the dependencies from PgpCore in version 5.5 and it's using BouncyCastle in version 1.8.9, and the issue still occurs. I even manually installed BouncyCastle to my project and it still throws the exception with the message that it cannot access the closed stream. Maybe I am doing something wrong in here? 

```     
        public static async Task<MemoryStream> EncryptFileStreamToStream()
        {
            var publicKeyString = await File.ReadAllTextAsync(@"C:\temp\Keys\public.asc");
            var publicKeyMemoryStream = new MemoryStream();
            await using var streamWriter = new StreamWriter(publicKeyMemoryStream, null, -1, true);
            await streamWriter.WriteAsync(publicKeyString);
            await streamWriter.FlushAsync();
            publicKeyMemoryStream.Position = 0;

            var content = await File.ReadAllTextAsync(@"C:\TEMP\Content\content.txt");
            var contentMemoryStream = new MemoryStream();
            await using var streamWriter2 = new StreamWriter(contentMemoryStream, null, -1, true);
            await streamWriter2.WriteAsync(content);
            await streamWriter2.FlushAsync();
            contentMemoryStream.Position = 0;

            EncryptionKeys encryptionKeys = new(publicKeyMemoryStream);
            using PGP pgp = new(encryptionKeys);
            MemoryStream ms = new();
            await pgp.EncryptStreamAsync(contentMemoryStream, ms);
            ms.Position = 0;
            contentMemoryStream.Position = 0;
            return ms;
        }
```

```
        public static async Task<MemoryStream> DecryptFileFromStream(MemoryStream enryptedContent)
        {
            var privateKey = await File.ReadAllTextAsync(@"C:\temp\Keys\private.asc");
            var privateKeyMemoryStream = new MemoryStream();
            await using var streamWriter = new StreamWriter(privateKeyMemoryStream, null, -1, true);
            await streamWriter.WriteAsync(privateKey);
            await streamWriter.FlushAsync();
            privateKeyMemoryStream.Position = 0;

            EncryptionKeys encryptionKeys = new(privateKeyMemoryStream, "password");
            using PGP pgp = new(encryptionKeys);
            MemoryStream ms = new();
            await pgp.DecryptStreamAsync(enryptedContent, privateKeyMemoryStream); //exception is thrown here
            ms.Position = 0;
            return ms;
        }
```

Usage: 
```
            var stream = await EncryptFileStreamToStream();
            await DecryptFileFromStream(stream);
```

**Comment by virouzrx** [comment]
That's right, I had a small mistake in code, seems like it works just fine right now. Thanks for the quick reply!

**Comment by etkirsch10** [comment]
Rolled back PGPCore to 5.4 and Bouncy Castle to 1.8.9 and worked perfectly.

Really useful wrapper - thx!

**Comment by sadatmazumder** [comment]
none of the following combinations works!:
- Rolled back PGPCore to 5.4 and Bouncy Castle to 1.8.9 
- Rolled back PGPCore to 5.4 and Bouncy Castle to 1.8.10

**Comment by robert-shattock** [comment]
FYI, have updated to the current latest version i.e. PGPCore 5.9.0 which now uses Portable.BouncyCastle 1.9.0 and can confirm I no longer get the error.  Thanks @mattosaurus for the fix somewhere after PGPCore 5.4.0.

**Comment by rakeshkumargupta** [comment]
I am still having issues with Portable.BouncyCastle 1.9.0. It's only working with Portable.BouncyCastle 1.8.9.
